src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +379,168 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Check the summary of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    public void validateSummaryTag(DetailNode ast) {
+        if (getInlineSummary(ast).isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            final String inlineSummaryDoc = getInlineSummary(ast);
+            if (isPeriodAtEnd(inlineSummaryDoc, period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+            }
+            else if (isSummaryValid(ast)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+            }
+        }
+    }
+
+    /**
+     * Checks if the string ends with period.
+     *
+     * @param sentence string to check for period at end.
+     * @param period String to check within sentence.
+     * @return true, is sentence ends with period.
+     */
+    private static boolean isPeriodAtEnd(String sentence, String period) {
+        final String summarySentence = sentence.trim();
+        return summarySentence.lastIndexOf(period) != summarySentence.length() - 1;
+    }
+
+    /**
+     * Get summary sentence for inline type summary javadoc.
+     *
+     * @param javadoc javadoc root node.
+     * @return  Summary sentence of javadoc root node.
+     */
+    private static String getInlineSummary(DetailNode javadoc) {
+        final String summary;
+        final DetailNode descriptionNode = getDescriptionNode(javadoc);
+        if (containsSummaryTagWithHtmlTag(javadoc)) {
+            summary = getInlineHtmlSentence(javadoc);
```

Comment:
what is reason to check html tags .... but in final ELSE block we do `summary = removeHtmlTagsFromString(summarySentence);`

Reply:
This is an parsing error of checkstyle. It parses a method like 
```
    /**
     * {@summary first doesn't have period
     * Use of html tags:
     * <ul>
     * <li>Item one.</li>
     * <li>Period here.</li>
     * </ul>}
     */
    private void invalidInlineJavadocTwo() // violation
    {
    }
```
 to
```
    |   |   |   |       |--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG [3:7]
    |   |   |   |       |   |--JAVADOC_INLINE_TAG_START -> { [3:7]
    |   |   |   |       |   |--CUSTOM_NAME -> @summary [3:8]
    |   |   |   |       |   |--WS ->   [3:16]
    |   |   |   |       |   |--DESCRIPTION -> DESCRIPTION [3:17]
    |   |   |   |       |   |   |--TEXT -> first doesn't have period [3:17]
    |   |   |   |       |   |   |--NEWLINE -> \n [3:42]
    |   |   |   |       |   |   |--LEADING_ASTERISK ->      * [4:0]
    |   |   |   |       |   |   |--TEXT ->  Use of html tags: [4:6]
    |   |   |   |       |   |   |--NEWLINE -> \n [4:24]
    |   |   |   |       |   |   |--LEADING_ASTERISK ->      * [5:0]
    |   |   |   |       |   |   |--TEXT ->  <ul> [5:6]
    |   |   |   |       |   |   |--NEWLINE -> \n [5:11]
    |   |   |   |       |   |   |--LEADING_ASTERISK ->      * [6:0]
    |   |   |   |       |   |   |--TEXT ->  <li>Item one.</li> [6:6]
    |   |   |   |       |   |   |--NEWLINE -> \n [6:25]
    |   |   |   |       |   |   |--LEADING_ASTERISK ->      * [7:0]
    |   |   |   |       |   |   |--TEXT ->  <li>Period here.</li> [7:6]
    |   |   |   |       |   |   |--NEWLINE -> \n [7:28]
    |   |   |   |       |   |   |--LEADING_ASTERISK ->      * [8:0]
    |   |   |   |       |   |   `--TEXT ->  </ul> [8:6]
    |   |   |   |       |   `--JAVADOC_INLINE_TAG_END -> } [8:12]
```

where as the `<ul>` should be parsed as an html element. This is the reason we use `removeHtmlTagsFromString(summarySentence)`

Should I make an issue for it?

Reply:
Is this the issue #9690 you already created?

Reply:
No Sir, this is something new. Earlier it failed to parse, in this one, it parses incorrectly. Here, after `Use of html tags:` the rest portion is to be parsed as an `HTML ELEMENT`, should I move ahead and create and issue, I will make all logs available there.

Reply:
@AkMo3 , please create issue on this to parse "ul" as HTML_ELEMENT.

Reply:
Already done at https://github.com/checkstyle/checkstyle/issues/9703